aries_cloudagent/transport/inbound/queue/redis.py

```diff
@@ -0,0 +1,140 @@
+"""Redis inbound transport."""
+import aioredis
+import asyncio
+import msgpack
+import logging
+
+from threading import Thread
+from urllib.parse import urlparse, ParseResult
+
+from ....core.profile import Profile
+
+from ...wire_format import DIDCOMM_V0_MIME_TYPE, DIDCOMM_V1_MIME_TYPE
+
+from ..manager import InboundTransportManager
+
+from .base import BaseInboundQueue, InboundQueueConfigurationError, InboundQueueError
+
+
+class RedisInboundQueue(BaseInboundQueue, Thread):
```

Comment:
Curious if you'd be willing to elaborate on your reasoning for running the inbound queue on a separate thread.